src/System.Text.Json/src/Resources/Strings.resx

```diff
@@ -0,0 +1,198 @@
+            </xsd:complexType>
+          </xsd:element>
+          <xsd:element name="resheader">
+            <xsd:complexType>
+              <xsd:sequence>
+                <xsd:element name="value" type="xsd:string" minOccurs="0" msdata:Ordinal="1" />
+              </xsd:sequence>
+              <xsd:attribute name="name" type="xsd:string" use="required" />
+            </xsd:complexType>
+          </xsd:element>
+        </xsd:choice>
+      </xsd:complexType>
+    </xsd:element>
+  </xsd:schema>
+  <resheader name="resmimetype">
+    <value>text/microsoft-resx</value>
+  </resheader>
+  <resheader name="version">
+    <value>2.0</value>
+  </resheader>
+  <resheader name="reader">
+    <value>System.Resources.ResXResourceReader, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <resheader name="writer">
+    <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <data name="ArrayDepthTooLarge" xml:space="preserve">
+    <value>Depth of {0} within an array is larger than max depth of {1}.</value>
+  </data>
+  <data name="ArrayEndWithinObject" xml:space="preserve">
+    <value>We are within an object but observed an ']'.</value>
+  </data>
+  <data name="Default" xml:space="preserve">
+    <value>An error occurred while reading the json.</value>
+  </data>
+  <data name="DepthMustBePositive" xml:space="preserve">
+    <value>Mismatched number of start/end objects or arrays. Depth is {0} but must be greater than 0.</value>
+  </data>
+  <data name="EndOfCommentNotFound" xml:space="preserve">
+    <value>Expected either a '\n' or '*/' to indicate end of comment, but instead reached end of data.</value>
+  </data>
+  <data name="EndOfStringNotFound" xml:space="preserve">
+    <value>Expected a '"' to indicate end of string, but instead reached end of data.</value>
+  </data>
+  <data name="ExpectedDigitNotFound" xml:space="preserve">
+    <value>Invalid number. Last character read: '{0}'. Expected a digit ('0'-'9').</value>
+  </data>
+  <data name="ExpectedDigitNotFoundEndOfData" xml:space="preserve">
+    <value>Invalid number. Last character read: '{0}'. Expected a digit ('0'-'9') but reached end of data.</value>
+  </data>
+  <data name="ExpectedEndAfterSingleJson" xml:space="preserve">
+    <value>Expected end of json after a single value but additional data found. Last character read: '{0}'.</value>
+  </data>
+  <data name="ExpectedEndOfDigitNotFound" xml:space="preserve">
+    <value>Invalid end of number. Last character read: '{0}'. Expected a delimiter.</value>
+  </data>
+  <data name="ExpectedFalse" xml:space="preserve">
+    <value>Expected a 'false' value, but instead got '{0}'.</value>
+  </data>
+  <data name="ExpectedNextDigitComponentNotFound" xml:space="preserve">
+    <value>Invalid number. Last character read: '{0}'. Expected '.' or 'E' or 'e'.</value>
```

Comment:
Having not seen the code, I don't understand when I'd get this vs the "I wanted a digit" message.  Comma and whitespace still also seem like they'd be valid, making the message misleading.

My general feedback (this is the last one) is to drop all of the "Expected" parts.  After all, no one has ever been helped by a compiler telling them `Expected: ;` :smile:

Reply:
Perhaps I'm not following, but certainly when a tool says "expected A, B, or C" I have often found that helpful.

Reply:
> but certainly when a tool says "expected A, B, or C" I have often found that helpful.

Yes, that's my thinking as well so I will try to keep the "expected ..." segment where possible. However, for this **particular** case, the "expected..." sentence fragment is incorrect and shouldn't be used. There is no number value where I can know that we expect a '.' or 'E' or 'e' (exclusively) next. We could have reached an end of a valid number and expect the next token.

That is why I need to fix this instance, at least. I will likely remove this exception message.